test/main.tf

```diff
@@ -0,0 +1,72 @@
+provider "aws" {
+  region  = "us-west-2"
+  profile = "thunderdome"
+}
+
+resource "random_string" "this" {
+  length = 16
+}
+
+resource "aws_cloudwatch_log_group" "group" {
+  name = "test-subscription-${random_string.this.id}"
+}
+
+module "cloudwatch_kinesis_firehose_tf" {
+  source           = "github.com/observeinc/terraform-aws-kinesis-firehose" # tflint-ignore: terraform_module_pinned_source
+  name             = "test-subscription-tf-${random_string.this.id}"
+  observe_customer = "101"
+  observe_token    = "fake:token"
+}
+
+module "cloudwatch_logs_subscription" {
+  source           = "../"
+  name             = "test-subscription-tf-${random_string.this.id}"
+  filter_name      = "test-tf-${random_string.this.id}"
+  kinesis_firehose = module.cloudwatch_kinesis_firehose_tf
+}
+
+resource "aws_iam_role" "subscription_filter" {
+  name               = "test-subscription-${random_string.this.id}"
+  assume_role_policy = <<-EOF
```

Comment:
Prefer [`data.aws_iam_policy_document`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) over embedded JSON:

* Terraform can perform structural validation via `terraform validate`
* Users get attribute completion from the provider
* It's easier to interpolate and avoid formatting mistakes

That said, a static policy like this is where the IAM policy document data source has the least impact.

Reply:
Sounds good, I see how "terraform validate" can speed up dev a lot more than the overhead of learning how to use the Data Source.

I think this can be a larger JIRA since we do this in a lot of places in our terraform.

Reply:
here's the JIRA, if you're interested in a task that involves going through the commit process 4x